torch/profiler/_pattern_matcher.py

```diff
@@ -603,8 +631,31 @@ def report_all_anti_patterns(prof, should_benchmark: bool = False, json_report_d
             if report_msg not in reported:
                 message_list.append(report_msg)
                 reported.add(report_msg)
+                src_location, line_no = source_code_location(event).split(":")
+                report_dict.setdefault(src_location, []).append({
+                    "line_number":
+                    int(line_no),
+                    "name":
+                    anti_pattern.name,
+                    "url":
+                    anti_pattern.url,
+                    "message":
+                    anti_pattern.description
+                })
+
+    if json_report_dir is not None:
+        json_report_path = os.path.join(json_report_dir,
+                                        "torchtidy_report.json")
+        if os.path.exists(json_report_path):
```

Comment:
It seems a bit strange to me to append by default. Maybe emit a file per run? (Or will that cause issues with loading into VSCode?)

Reply:
Yeah it definitely be difficult to have multiple files. If we have two different json for the same file, we don't know which to look at (difficult to handle). Also, having a single file helps me update the problems tab. I only need to setup a change listener to 1 file, instead of constantly searching for new files & add to listeners.

Reply:
Gotcha.